core/src/main/java/io/opentelemetry/android/features/diskbuffering/scheduler/ExportScheduleAutoDetector.kt

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.features.diskbuffering.scheduler
+
+import android.app.ActivityManager
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.os.BatteryManager
+import android.util.Log
+import androidx.annotation.VisibleForTesting
+import io.opentelemetry.android.common.RumConstants.OTEL_RUM_LOG_TAG
+
+/**
+ * Auto-detects optimal telemetry export frequency based on device conditions.
+ *
+ * This class analyzes device state and suggests an appropriate export interval:
+ * - Devices under memory pressure: Increase interval (export less frequently)
+ * - Devices on battery (not charging): Increase interval (preserve battery)
+ * - Devices with low battery: Increase interval significantly
+ * - Normal conditions: Use default interval
+ *
+ * The auto-detected value can be overridden by explicitly setting
+ * [DiskBufferingConfig.exportScheduleDelayMillis].
+ */
+internal object ExportScheduleAutoDetector {
+    private const val DEFAULT_EXPORT_INTERVAL_MILLIS = 10000L // 10 seconds
+    private const val BATTERY_SAVER_INTERVAL_MILLIS = 30000L // 30 seconds
+    private const val LOW_MEMORY_INTERVAL_MILLIS = 20000L // 20 seconds
```

Comment:
```suggestion
    private const val DEFAULT_EXPORT_INTERVAL_MILLIS = 10000L
    private const val BATTERY_SAVER_INTERVAL_MILLIS = 30000L
    private const val LOW_MEMORY_INTERVAL_MILLIS = 20000L
```
this should not be neccessart `// 10 seconds` as `10000L` itsefl implies the same thing